surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,360 @@
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
+Type: select one
+
+- I have never used a debugger
+- I am uncomfortable using a debugger
+- I am comfortable using a debugger
+- I am very comfortable using a debugger
+
+> *Justification:*
+>
+> Users who are more familiar with debuggers will face different problems than
+> users who are not.
+
+### Why are you uncomfortable using a debugger?
+If: uncomfortable with debugger
+Type: select all that apply
+
+- Unfamiliar (just need to spend more time)
+- Reliability (don't expect debugger to work well enough)
+- Ease-of-use
+- Lack of documentation with Rust
+
+> *Justification:*
+>
+> Why aren't users comfortable with debuggers is useful.
+
+### Do you often use a debugger with other programming languages (anything but Rust)?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> If a user regularly uses a debugger for other languages but not for Rust then
+> it would be interesting to know that.
+
+### Which programming language(s) do you use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- Go
+- C#
+
+> *Justification:*
+>
+> Debugging experiences vary widely by language - debugging a interpreted
+> language like Python is very different from a compiled language like C/C++ -
+> this will inform user expectations with Rust debuggers.
+
+### How does debugging in language X compare to Rust?
+Type: freeform text
+
+> *Justification:*
+>
+> What can Rust's debugging experience learn from other languages.
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
```

Comment:
Do we need the previous question if we ask this one?

Reply:
I think it's helpful to know specifically how a user is interfacing with their debugger because I think that will affect their experience considerably (I'm not sure what actions we might take differently in response to this information - maybe just changing recommendations we make).

Reply:
What I mean is, presumably all the information we gain from the previous question is also gained from this question. If someone says they use the lldb vs code extension, we know they use lldb. So we can either get rid of the previous question or make this question more general

### Which debugger interfaces do you use with Rust?

* Editor Extension
* Built-in IDE debugger
* terminal line interface
* terminal UI 
* Separate debugging program

Reply:
That makes sense, changed it to what you have suggested.